File: Backendspr/src/main/java/com/example/backendspr/Models/Diet.java
package com.example.backendspr.Models;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
@Table( name = "Diet")
public class Diet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String description;

    @Enumerated(EnumType.STRING)
    private DietType type;

    @ManyToMany
    private List<Food> foods;


}

File: FrontendAng/src/app/Core/Models/diet.ts
import {Food} from "./food";

export interface Diet {
  id: number;
  name: string;
  description: string;
  type: DietType;
  prix:number;
  foods: Food[]; // Array of Food IDs
}

export enum DietType {
  PROTEIN = 'PROTEIN',
  CARB = 'CARB',
  FAT = 'FAT'
}
